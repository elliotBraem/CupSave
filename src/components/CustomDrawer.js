/* eslint-disable react/prop-types */
import React, {Component} from 'react';
import {Text, StyleSheet, View, Image, TouchableOpacity} from 'react-native';
import {DrawerItems, SafeAreaView} from 'react-navigation';
import {connect} from 'react-redux';
import PropTypes from 'prop-types';
import Logout from '../assets/images/drawer-icons/logout-icon.svg';
import COLORS from '../constants/colors';
import * as authActions from '../store/actions/auth';

const profileImage = require('../assets/images/profileicon.png');

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  headerContainer: {
    marginLeft: 20,
    marginBottom: 20,
  },
  headerText: {
    fontSize: 14,
    color: COLORS.white,
  },
  profileImage: {
    width: 80,
    height: 80,
    marginBottom: 20,
    borderRadius: 40,
  },
  bottom: {
    justifyContent: 'flex-end',
    flex: 1,
    marginBottom: 36,
    marginLeft: 20,
  },
  innerBottom: {
    flexDirection: 'row',
  },
  logout: {
    marginLeft: 10,
    color: COLORS.white,
    fontSize: 16,
  },
});

class CustomDrawer extends Component {
  static propTypes = {
    navigation: PropTypes.shape({
      navigate: PropTypes.func.isRequired,
    }).isRequired,
    auth: PropTypes.shape({
      isAuthenticated: PropTypes.bool.isRequired,
    }).isRequired,
    logout: PropTypes.func.isRequired,
  };

  state = {errorMessage: null};

  handleLogout = async () => {
    const {logout, navigation} = this.props;
    await logout()
      .then(navigation.navigate('Auth'))
      .catch(error => {
        console.log(error);
        this.setState({errorMessage: error});
      });
  };

  render() {
    const {logout, auth, ...props} = this.props;
    return (
      <SafeAreaView style={styles.container} forceInset={{top: 'always', horizontal: 'never'}}>
        <View style={styles.headerContainer}>
          <Image source={profileImage} style={styles.profileImage} />
          <Text style={styles.headerText}>{auth.user.email}</Text>
        </View>
        <View>
          <DrawerItems {...props} />
        </View>
        <View style={styles.bottom}>
          <TouchableOpacity onPress={this.handleLogout} style={styles.innerBottom}>
            <Logout style={{width: 20, height: 20}} />
            <Text style={styles.logout}>Log out</Text>
          </TouchableOpacity>
        </View>
      </SafeAreaView>
    );
  }
}

const mapDispatchToProps = (dispatch, ownProps) => {
  return {
    logout: () => dispatch(authActions.dbLogout()),
  };
};

const mapStateToProps = (state, ownProps) => {
  const auth = state.auth || {};
  return {
    auth,
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(CustomDrawer);
