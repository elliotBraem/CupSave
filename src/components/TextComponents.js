import React from 'react';
import {Text, StyleSheet} from 'react-native';
import COLORS from '../constants/colors';

const styles = StyleSheet.create({
  normalText: {
    fontFamily: 'open-sans-regular',
    fontSize: 22,
    color: COLORS.white,
  },
  statText: {
    fontSize: 24,
    color: COLORS.secondary,
  },
  statSubtext: {
    fontSize: 12,
  },
  titleText: {
    fontSize: 16,
  },
  headerTitle: {
    fontSize: 20,
    color: COLORS.white,
  },
  worldCounterText: {
    fontFamily: 'open-sans-italic',
    color: COLORS.white,
    fontSize: 18,
    marginTop: 20,
  },
  saveACupText: {
    fontFamily: 'open-sans-bold',
    marginTop: 40,
    color: COLORS.primary,
    fontSize: 48,
    paddingRight: 60,
  },
  saveTheWorldText: {
    fontFamily: 'open-sans-italic',
    color: COLORS.white,
    fontSize: 48,
    paddingLeft: 40,
  },
  aboutUsText: {
    fontFamily: 'open-sans',
    color: COLORS.white,
    fontSize: 20,
    paddingLeft: 10,
  },
  aboutUsAuthorText: {
    fontFamily: 'open-sans-italic',
    color: COLORS.secondary,
    fontSize: 14,
    paddingLeft: 10,
  },
});

export const AppText = props => {
  return <Text style={styles.normalText}>{props.children}</Text>;
};

export const StatText = props => {
  return <Text style={styles.statText}>{props.children}</Text>;
};

export const StatSubtext = props => {
  return <Text style={styles.statSubtext}>{props.children}</Text>;
};

export const TitleText = props => {
  return <Text style={styles.titleText}>{props.children}</Text>;
};

export const HeaderTitle = props => {
  return <Text style={styles.headerTitle}>{props.children}</Text>;
};

export const WorldCounterText = props => {
  return <Text style={styles.worldCounterText}>{props.children}</Text>;
};

export const SaveACupText = props => {
  return <Text style={styles.saveACupText}>{props.children}</Text>;
};

export const SaveTheWorldText = props => {
  return <Text style={styles.saveTheWorldText}>{props.children}</Text>;
};

export const AboutUsText = props => {
  return <Text style={styles.aboutUsText}>{props.children}</Text>;
};

export const AboutUsAuthorText = props => {
  return <Text style={styles.aboutUsAuthorText}>{props.children}</Text>;
};
