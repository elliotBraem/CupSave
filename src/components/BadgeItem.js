import React from 'react';
import {Image, StyleSheet, View} from 'react-native';
import PropTypes from 'prop-types';
import {AppText} from './TextComponents';
import COLORS from '../constants/colors';

const styles = StyleSheet.create({
  container: {
    flexDirection: 'column',
    alignItems: 'center',
    marginRight: 20,
  },
  circle: {
    width: 60,
    height: 60,
    borderRadius: 30,
  },
});

const BadgeItem = ({name, icon}) => (
  <View style={styles.container}>
    <Image style={styles.circle} source={{uri: icon}} />
    {/* <AppText>{name}</AppText> */}
  </View>
);

BadgeItem.propTypes = {
  name: PropTypes.string.isRequired,
  icon: PropTypes.string.isRequired,
};

export default BadgeItem;
