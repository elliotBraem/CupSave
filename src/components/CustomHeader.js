import React from 'react';
import {StyleSheet, View, Platform} from 'react-native';
import PropTypes from 'prop-types';
import {compose} from 'recompose';
import {withNavigation} from 'react-navigation';
import BurgerIcon from '../assets/images/drawer-icons/burger-icon.svg';
import {HeaderTitle} from './TextComponents';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: 'transparent',
    width: '100%',
    height: Platform.OS === 'ios' ? 70 : 70 - 24,
    marginTop: Platform.OS === 'ios' ? 30 : 30 - 24,
    alignContent: 'center',
    flexDirection: 'row',
  },
  burgerIcon: {
    width: 30,
    height: 30,
    marginLeft: 20,
    marginRight: 20,
  },
});

const CustomHeader = ({navigation, title}) => {
  return (
    <View style={styles.container}>
      <BurgerIcon style={styles.burgerIcon} onPress={() => navigation.toggleDrawer()} />
      <HeaderTitle style={styles.headerTitle}>{title}</HeaderTitle>
    </View>
  );
};

CustomHeader.propTypes = {
  navigation: PropTypes.shape({
    toggleDrawer: PropTypes.func.isRequired,
  }).isRequired,
  title: PropTypes.string.isRequired,
};

const enhance = compose(withNavigation);

export default enhance(CustomHeader);
