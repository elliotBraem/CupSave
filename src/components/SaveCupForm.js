import React from 'react';
import {View, StyleSheet, Image, TouchableOpacity} from 'react-native';
import PropTypes from 'prop-types';
import {withHandlers, compose} from 'recompose';
import {connect} from 'react-redux';
import {withFirebase, withFirestore} from 'react-redux-firebase';
import COLORS from '../constants/colors';
import {AppText} from './TextComponents';
import Logo from '../assets/images/logo.png';

const styles = StyleSheet.create({
  container: {
    width: 300,
    height: 130,
    borderRadius: 12,
    backgroundColor: COLORS.white,
    alignItems: 'center',
  },
  button: {
    marginTop: 10,
    width: 270,
    height: 70,
    borderRadius: 12,
    backgroundColor: COLORS.primary,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
  },
  logoIcon: {
    height: 45,
    width: 30,
    marginRight: 30,
  },
});

const SaveCupForm = ({onSaveCupFormSubmit}) => (
  <View style={styles.container}>
    <TouchableOpacity onPress={onSaveCupFormSubmit} style={styles.button}>
      <Image source={Logo} style={styles.logoIcon} />
      <AppText>SAVE A CUP</AppText>
    </TouchableOpacity>
  </View>
);

SaveCupForm.propTypes = {
  onSaveCupFormSubmit: PropTypes.func.isRequired,
};

const enhance = compose(
  withFirebase,
  withFirestore,
  connect(({firebase: {auth}}) => ({
    auth,
  })),
  withHandlers({
    onSaveCupFormSubmit: props => () => {
      const currentUID = props.auth.uid;
      const userRef = props.firestore.collection('users').doc(`${currentUID}`);
      props.firestore
        .runTransaction(async transaction => {
          const doc = await transaction.get(userRef);

          const currentTimeInUnixEpoch = new Date().valueOf();

          const newTotal = doc.exists ? doc.data().consumption.total + 1 : 1;

          const data = {
            consumption: {
              total: newTotal,
              most_recent_consumption: currentTimeInUnixEpoch,
              history: {
                [currentTimeInUnixEpoch]: newTotal,
              },
            },
          };

          transaction.update(userRef, data);
        })
        .catch(error => {
          this.setState({errorMessage: error.message});
        });
    },
  })
);

export default enhance(SaveCupForm);
