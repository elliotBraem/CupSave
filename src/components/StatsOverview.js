import React from 'react';
import {View, StyleSheet} from 'react-native';
import PropTypes from 'prop-types';
import COLORS from '../constants/colors';
import {StatText} from './TextComponents';

const styles = StyleSheet.create({
  container: {
    width: 350,
    height: 90,
    backgroundColor: COLORS.white,
    borderRadius: 12,
    flexDirection: 'row',
    justifyContent: 'space-evenly',
    alignItems: 'center',
  },
});

const StatsOverview = ({drinkSize, totalCupsSaved, level}) => (
  <View style={styles.container}>
    <StatText style={styles.statText}>{drinkSize}</StatText>
    <StatText style={styles.statText}>{totalCupsSaved}</StatText>
    <StatText style={styles.statText}>{level}</StatText>
  </View>
);

StatsOverview.propTypes = {
  drinkSize: PropTypes.number,
  totalCupsSaved: PropTypes.number,
  level: PropTypes.number,
};

StatsOverview.defaultProps = {
  drinkSize: 16,
  totalCupsSaved: 0,
  level: 0,
};

export default StatsOverview;
