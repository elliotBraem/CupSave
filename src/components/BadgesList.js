import React, {Component} from 'react';
import {View, StyleSheet, FlatList} from 'react-native';
import PropTypes from 'prop-types';
import {withFirebase, withFirestore} from 'react-redux-firebase';
import {compose} from 'redux';
import {connect} from 'react-redux';
import BadgeItem from './BadgeItem';

const styles = StyleSheet.create({
  container: {
    width: 320,
    height: 60,
    flexDirection: 'row',
    justifyContent: 'space-evenly',
  },
});

class BadgesList extends Component {
  renderItem = ({item}) => <BadgeItem {...item} />;

  keyExtractor = item => item.key;

  render() {
    const {...props} = this.props;
    // console.log(data);
    return <FlatList keyExtractor={this.keyExtractor} renderItem={this.renderItem} {...props} />;
  }
}

export default BadgesList;
