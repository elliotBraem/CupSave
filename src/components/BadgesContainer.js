import React from 'react';
import {View, StyleSheet} from 'react-native';
import PropTypes from 'prop-types';
import BadgesList from './BadgesList';
import {TitleText} from './TextComponents';
import COLORS from '../constants/colors';

const styles = StyleSheet.create({
  container: {
    width: '95%',
    height: 130,
    backgroundColor: COLORS.white,
    borderRadius: 12,
    flexDirection: 'column',
    justifyContent: 'space-between',
    shadowColor: COLORS.secondary,
    shadowOffset: {
      width: 3,
      height: 3,
    },
    shadowOpacity: 0.3,
    shadowRadius: 12,
    marginTop: 15,
  },
  inner: {
    marginTop: 15,
    marginLeft: 15,
  },
  title: {
    marginLeft: 5,
  },
  list: {
    paddingTop: 10,
  },
});

const BadgesContainer = ({badges}) => (
  <View style={styles.container}>
    <View style={styles.inner}>
      <View style={styles.title}>
        <TitleText>My Badges</TitleText>
      </View>
      <BadgesList style={styles.list} data={badges.badgeList} />
    </View>
  </View>
);

BadgesContainer.propTypes = {
  badges: PropTypes.shape({
    badgeList: PropTypes.array.isRequired,
  }).isRequired,
};

export default BadgesContainer;
