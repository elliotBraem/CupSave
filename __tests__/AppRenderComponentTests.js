import React from 'react';
import renderer from 'react-test-renderer';

import {CustomDrawer} from '../src/components/CustomDrawer';
import {CustomHeader} from '../src/components/CustomHeader';
import LoadingComp from '../src/components/Loading';
import SaveCupForm from '../src/components/SaveCupForm';
import {AppText, StatText, StatSubtext, TitleText, HeaderTitle} from '../src/components/TextComponents';
import StatsOverview from '../src/components/StatsOverview';

// Components -> CustomDrawer
it('CustomDrawer renders without crashing', () => {
  console.error = jest.fn();
  const rendered = renderer.create(<CustomDrawer auth={{user: {email: 'test@test.com'}}} />).toJSON();
  expect(rendered).toBeTruthy();
  expect(console.error).toHaveBeenCalled();
});

it('CustomDrawer test against snapshot', () => {
  const tree = renderer.create(<CustomDrawer auth={{user: {email: 'test@test.com'}}} />).toJSON();
  expect(tree).toMatchSnapshot();
});

// Components -> CustomHeader
it('CustomHeader renders without crashing', () => {
  console.error = jest.fn();
  const rendered = renderer.create(<CustomHeader />).toJSON();
  expect(rendered).toBeTruthy();
  expect(console.error).toHaveBeenCalled();
});

it('CustomHeader test against snapshot', () => {
  const tree = renderer.create(<CustomHeader />).toJSON();
  expect(tree).toMatchSnapshot();
});

// Components -> Loading
it('Loading renders without crashing', () => {
  const rendered = renderer.create(<LoadingComp />).toJSON();
  expect(rendered).toBeTruthy();
});

it('Loading test against snapshot', () => {
  const tree = renderer.create(<LoadingComp />).toJSON();
  expect(tree).toMatchSnapshot();
});

// Components -> SaveCupForm
it('SaveCupForm renders without crashing', () => {
  const rendered = renderer.create(<SaveCupForm />).toJSON();
  expect(rendered).toBeTruthy();
});

it('SaveCupForm test against snapshot', () => {
  const tree = renderer.create(<SaveCupForm />).toJSON();
  expect(tree).toMatchSnapshot();
});

// Components -> StatsOverview
it('StatsOverview renders without crashing', () => {
  const rendered = renderer.create(<StatsOverview />).toJSON();
  expect(rendered).toBeTruthy();
});

it('StatsOverview test against snapshot', () => {
  const tree = renderer.create(<StatsOverview />).toJSON();
  expect(tree).toMatchSnapshot();
});

// Components -> TextComponents -> AppText
it('AppText renders without crashing', () => {
  const rendered = renderer.create(<AppText />).toJSON();
  expect(rendered).toBeTruthy();
});

it('AppText test against snapshot', () => {
  const tree = renderer.create(<AppText />).toJSON();
  expect(tree).toMatchSnapshot();
});

// Components -> TextComponents -> StatText
it('StatText renders without crashing', () => {
  const rendered = renderer.create(<StatText />).toJSON();
  expect(rendered).toBeTruthy();
});

it('StatText test against snapshot', () => {
  const tree = renderer.create(<StatText />).toJSON();
  expect(tree).toMatchSnapshot();
});

// Components -> TextComponents -> StatSubtext
it('StatSubtext renders without crashing', () => {
  const rendered = renderer.create(<StatSubtext />).toJSON();
  expect(rendered).toBeTruthy();
});

it('StatSubtext test against snapshot', () => {
  const tree = renderer.create(<StatSubtext />).toJSON();
  expect(tree).toMatchSnapshot();
});

// Components -> TextComponents -> TitleText
it('TitleText renders without crashing', () => {
  const rendered = renderer.create(<TitleText />).toJSON();
  expect(rendered).toBeTruthy();
});

it('TitleText test against snapshot', () => {
  const tree = renderer.create(<TitleText />).toJSON();
  expect(tree).toMatchSnapshot();
});

// Components -> TextComponents -> HeaderTitle
it('HeaderTitle renders without crashing', () => {
  const rendered = renderer.create(<HeaderTitle />).toJSON();
  expect(rendered).toBeTruthy();
});

it('HeaderTitle test against snapshot', () => {
  const tree = renderer.create(<HeaderTitle />).toJSON();
  expect(tree).toMatchSnapshot();
});
