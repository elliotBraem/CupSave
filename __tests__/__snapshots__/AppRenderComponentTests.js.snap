// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppText test against snapshot 1`] = `
<Text
  style={
    Object {
      "color": "#ffffff",
      "fontSize": 22,
    }
  }
/>
`;

exports[`HeaderTitle test against snapshot 1`] = `
<Text
  style={
    Object {
      "color": "#ffffff",
      "fontSize": 20,
    }
  }
/>
`;

exports[`Loading test against snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#79db85"
    hidesWhenStopped={true}
    size="large"
  />
</View>
`;

exports[`SaveCupForm test against snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ffffff",
      "borderRadius": 12,
      "height": 130,
      "shadowColor": "#1A1A1A",
      "shadowOffset": Object {
        "height": 3,
        "width": 3,
      },
      "shadowOpacity": 0.3,
      "shadowRadius": 12,
      "width": 300,
    }
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#79db85",
        "borderRadius": 12,
        "flexDirection": "row",
        "height": 70,
        "justifyContent": "center",
        "marginTop": 10,
        "opacity": 1,
        "shadowColor": "#1A1A1A",
        "shadowOffset": Object {
          "height": 3,
          "width": 3,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 12,
        "width": 270,
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 45,
          "marginRight": 30,
          "width": 30,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#ffffff",
          "fontSize": 22,
        }
      }
    >
      SAVE A CUP
    </Text>
  </View>
</View>
`;

exports[`StatSubtext test against snapshot 1`] = `
<Text
  style={
    Object {
      "fontSize": 12,
    }
  }
/>
`;

exports[`StatText test against snapshot 1`] = `
<Text
  style={
    Object {
      "color": "#1A1A1A",
      "fontSize": 24,
    }
  }
/>
`;

exports[`StatsOverview test against snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ffffff",
      "borderRadius": 12,
      "flexDirection": "row",
      "height": 90,
      "justifyContent": "space-evenly",
      "shadowColor": "#1A1A1A",
      "shadowOffset": Object {
        "height": 3,
        "width": 3,
      },
      "shadowOpacity": 0.3,
      "shadowRadius": 12,
      "width": "95%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "column",
        "height": "60%",
        "justifyContent": "space-between",
        "width": "33%",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 12,
        }
      }
    >
      My drink size
    </Text>
    <Text
      style={
        Object {
          "color": "#1A1A1A",
          "fontSize": 24,
        }
      }
    >
      16oz
    </Text>
  </View>
  <View
    style={
      Object {
        "borderLeftColor": "#d4d4d4",
        "borderLeftWidth": 1,
        "height": 70,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "column",
        "height": "60%",
        "justifyContent": "space-between",
        "width": "33%",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 12,
        }
      }
    >
      Total cups saved
    </Text>
    <Text
      style={
        Object {
          "color": "#1A1A1A",
          "fontSize": 24,
        }
      }
    >
      0
    </Text>
  </View>
  <View
    style={
      Object {
        "borderLeftColor": "#d4d4d4",
        "borderLeftWidth": 1,
        "height": 70,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "column",
        "height": "60%",
        "justifyContent": "space-between",
        "width": "33%",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 12,
        }
      }
    >
      Level
    </Text>
    <Text
      style={
        Object {
          "color": "#1A1A1A",
          "fontSize": 24,
        }
      }
    >
      0
    </Text>
  </View>
</View>
`;

exports[`TitleText test against snapshot 1`] = `
<Text
  style={
    Object {
      "fontSize": 16,
    }
  }
/>
`;
